paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSink.java

```diff
@@ -102,8 +103,9 @@ private StoreSinkWrite.Provider createWriteProvider(
             Options options = table.coreOptions().toConfiguration();
             ChangelogProducer changelogProducer = table.coreOptions().changelogProducer();
             waitCompaction =
-                    changelogProducer == ChangelogProducer.LOOKUP
-                            && options.get(CHANGELOG_PRODUCER_LOOKUP_WAIT);
+                    (changelogProducer == ChangelogProducer.LOOKUP
+                                    && options.get(CHANGELOG_PRODUCER_LOOKUP_WAIT))
+                            || options.get(DELETION_VECTORS_ENABLED);
```

Comment:
If `CHANGELOG_PRODUCER_LOOKUP_WAIT` is true, can we just not wait?

Reply:
Will support deletion vectors with lookup wait in the another PR